### PR TITLE
Use the default FreeBSD trust store according to certctl(8)

### PR DESCRIPTION
**Pull request by michael-o**
FreeBSD committer here; The use of `ca_root_nss` package has been long superseded with a default system-wide truststore used by OpenSSL, managed through `certctl(8)`. There is no need to rely on `ca_root_nss` package.

### CODE REVIEW

File: src/CaBundle.php [hunk 1]
```diff
@@ -84,13 +84,12 @@ public static function getSystemCaRootBundlePath(?LoggerInterface $logger = null
             '/etc/pki/tls/certs/ca-bundle.crt', // Fedora, RHEL, CentOS (ca-certificates package)
             '/etc/ssl/certs/ca-certificates.crt', // Debian, Ubuntu, Gentoo, Arch Linux (ca-certificates package)
             '/etc/ssl/ca-bundle.pem', // SUSE, openSUSE (ca-certificates package)
-            '/usr/local/share/certs/ca-root-nss.crt', // FreeBSD (ca_root_nss_package)
+            '/etc/ssl/certs', // FreeBSD
```
Comment (Seldaek) [inline src/CaBundle.php:87]: Is there any CA file name in there we should look at? Because as you can see below at line 100 we do add every file's directory anyway as an entry, so only listing the dir here will make it scan both `/etc/ssl/certs` AND `/etc/ssl` which is not what you want.

Comment (michael-o) [reply src/CaBundle.php:87]: You mean this, right:
https://github.com/composer/ca-bundle/blob/54498fa4b0cbb1f42c97541c5c862432ebf2745e/src/CaBundle.php#L99-L101

That if clause does not make any sense to me. You cannot make that assumption and I consider it as a waste of cycles to add and probe those. It is either a dir or a file. Looking at the default layout of `OPENSSLDIR`, it does not intersect unless you go explicit. There is only `/etc/ssl/certs` with hashed CA certs links are described by https://docs.openssl.org/3.3/man1/openssl-rehash/ on FreeBSD. Everything else is undefined/custom like the env vars or the `php.ini` entries.

In short: There is no CA bundle by default installed/configured with/by the system on FreeBSD.

Comment (michael-o) [reply src/CaBundle.php:87]: The source for the base system is here: https://github.com/freebsd/freebsd-src/tree/main/secure/caroot

Comment (michael-o) [reply src/CaBundle.php:87]: In fact, proper behavior was broken for me with https://github.com/composer/ca-bundle/commit/4a447c53c5b1c73c1fc71d3a169dde2dadb75c9d

Comment (Seldaek) [reply src/CaBundle.php:87]: Yeah I can imagine in some cases it breaks, and maybe in others it helps..

But anyway we should remove it from this line then, and add it below the foreach as a single entry that won't get the dirname treatment, if you see what I mean

Comment (michael-o) [reply src/CaBundle.php:87]: > Yeah I can imagine in some cases it breaks, and maybe in others it helps..
> 
> But anyway we should remove it from this line then, and add it below the foreach as a single entry that won't get the dirname treatment, if you see what I mean

Honestly not, can you rephrase or propose a patch with your idea?

Comment (michael-o) [reply src/CaBundle.php:87]: FWIW: I do remember that Debian has `/etc/ssl/certs` as well and RHEL changed it in a more recent version. At least @tiran told me so. See also: https://github.com/tiran/certifi-system-store/issues/3

Comment (Seldaek) [reply src/CaBundle.php]: Ok so now we're left with two removals, which really reduces compatibility with older FreeBSD. If you tell me that's fine I'll take your word for it, but I don't think it changes anything for modern versions.

File: src/CaBundle.php [hunk 2]
```diff
@@ -82,15 +82,13 @@ public static function getSystemCaRootBundlePath(?LoggerInterface $logger = null
 
         $otherLocations = array(
             '/etc/pki/tls/certs/ca-bundle.crt', // Fedora, RHEL, CentOS (ca-certificates package)
-            '/etc/ssl/certs/ca-certificates.crt', // Debian, Ubuntu, Gentoo, Arch Linux (ca-certificates package)
+            '/etc/ssl/certs/ca-certificates.crt', // Debian, Ubuntu, Gentoo, Arch Linux (ca-certificates package), FreeBSD (only the /etc/ssl/certs dir which gets added below)
```
Comment (michael-o) [inline src/CaBundle.php:85]: This really read awkward. It looks rather like a hack than a fix. Maybe a test in the look shall determine whether `otherLocations` entry is a dir or file?

File: src/CaBundle.php (hunk 1 above)
Comment (michael-o) [reply src/CaBundle.php]: I am confident because `certlctl(8)` has now be around for five years: https://github.com/freebsd/freebsd-src/commits/main/usr.sbin/certctl. Time to embrace ti.

File: src/CaBundle.php (hunk 2 above)
Comment (Seldaek) [reply src/CaBundle.php]: Ok I tweaked it but this will remove a few lookups, which I think is probably fine.. Hopefully it doesn't cause regressions.

Comment (michael-o) [reply src/CaBundle.php]: Thank you, merged appreciated. As soon as this lands in composer I will create a downstream PR to remove the depencency to ca_root_nss in FreeBSD.

Comment (michael-o) [reply src/CaBundle.php]: > Composer 2.8.3 ships with this PR

Magic, thank you! I will create the downstream request with the FreeBSD port to remove the dependency on ca_root_nss!

Comment (michael-o) [reply src/CaBundle.php]: Downstream request: https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=282816
